custom_components/rpi_waveshare_ups/__init__.py

```diff
@@ -56,15 +59,15 @@ def _close(self) -> None:
 
     def gather_details(self) -> None:
         """Retrieve the required details for the UPS."""
-        self._current = self._ina219.get_current_ma()
+        self._current = -self._ina219.get_current_ma()
```

Comment:
QQ: on this one... negating the value seems to be correct for the model D but I didn't see this happening for the model A or B in the example code provided by Waveshare.  Should this be negated for all models?

Reply:
ops. forget to change this one.... Updated. Thanks